crypto/ui/ui_util.c

```diff
@@ -108,15 +108,14 @@ static int ui_read(UI *ui, UI_STRING *uis)
     case UIT_PROMPT:
         {
             char result[PEM_BUFSIZE + 1];
+            memset(result, 0, sizeof(result));
```

Comment:
Another way would be an explicit initializing assignment:

``` C
            char result[PEM_BUFSIZE + 1] = { 0, };
```

That would avoid the variable declaration / statement mix, which OpenSSL coding style doesn't allow

Reply:
Apart from a missing blank line betwixt the declaration and code, this conforms to our style.

Reply:
@bernd-edlinger @levitte @paulidale You are more familiar with code. Which fix are more applicable? With initializing initial buffer with zero or with changing default return value in UI_process()? What do you think? I suppose, changes in Ui_process can be dangerous, I don't won't to break something

Reply:
> Another way would be an explicit initializing assignment:
> 
> ```c
>             char result[PEM_BUFSIZE + 1] = { 0, };
> ```
> 
> That would avoid the variable declaration / statement mix, which OpenSSL coding style doesn't allow

FIxed

Reply:
So, I need to return an error from ui process in case OPENSSL_NO_UI_CONSOLE macro defined?

Reply:
yes, or if the NULL UI method is used by any reason.

Reply:
BTW: I just skimmed a similar issue in `UI_UTIL_read_pw`:
where 0 is returned, and the password string is undefined as well if
a UI_new returns NULL due to out-of-memory error.